GameLogic/engine/StandardGame.cs

```diff
@@ -6,180 +6,139 @@
-        /// </returns>  
-        public StandardGameState DeserializeFromFEN(string fenNotation)
+        public StandardGame(string fenNotation)
         {
-            var splitFenNotation = fenNotation.Split(' ');
-
-            if (splitFenNotation.Count() != 6)
-                throw new ArgumentException("Invalid FEN notation.");
-
-            var gameState = new StandardGameState(
-                SerializeHelper.NotationToBoard(splitFenNotation[0]),
-                SerializeHelper.NotationToColor(splitFenNotation[1]),
-                SerializeHelper.NotationToCastle(splitFenNotation[2]),
-                SerializeHelper.NotationToCell(splitFenNotation[3]),
-                Int32.Parse(splitFenNotation[4]),
-                Int32.Parse(splitFenNotation[5])
-            );
-            return gameState;
+            this.gameState = StandardFENSerializer.DeserializeFromFEN(fenNotation);
         }
 
+        public StandardGame(StandardGameState gameState)
+        {
+            this.gameState = gameState;
+        }
 
-        /// <summary>
-        /// TODO: not finished.
-        /// </summary>
-        public bool IsMate(StandardGameState gameState)
+        // Checks if the mate occurs at the current game state.
+        // Enemy color mates/wins the active color.
+        // 
+        // Returns
+        // -------
+        // true, if the mate occurs. Otherwise, false.
+        public bool IsMate()
         {
-            foreach (var move in FindAllMoves(gameState))
-            {
-                var nextGameState = MakeMove(gameState, move);
-                if (!IsCheck(nextGameState))
-                {
-                    return false;
-                }
-            }
-            return true;
+            return IsCheck()
+                   && FindAllValidMoves().Count == 0;
         }
-        /// <summary>
-        /// TODO: not finished.
-        /// </summary>
-        public bool IsCheck(StandardGameState gameState)
+
+        // Checks if the check occurs at the current game state.
+        // Enemy color checks the active color.
+        //
+        // Returns
+        // -------
+        // true, if the check occurs. Otherwise, false.
+        public bool IsCheck()
         {
-            // Extract king location(s). 
-            var kingCells = gameState.Board.GetCellsWithPieces(filterByColor: gameState.ActiveColor,
-                                                               filterByPieceType: PieceType.King);
-            if (kingCells.Count != 1)
-            {
-                throw new InvalidOperationException("Invalid game state. There is only one king on standard board.");
-            }
 
-            // Extract only one king cell.
-            var kingCell = kingCells[0];
+            var kingCell = gameState.Board.GetCellsWithPieces(filterByColor: gameState.ActiveColor,
+                                                              filterByPieceType: PieceType.King) // Extract king location.
```

Comment:
The comment `// Extract king location` should go before this whole expression.
This way it looks like some extraction happens on line 56 exactly, which is wrong.

Reply:
Fixed!